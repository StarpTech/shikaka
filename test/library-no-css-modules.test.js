const { prepareLibraryTest } = require('./utils');

describe('library-no-css-modules fixture', () => {
  test('build with shikaka', async () => {
    const {
      files: { Button, styles }
    } = await prepareLibraryTest('library-no-css-modules', 'src/index.js', ['--no-cssmodules']);

    expect(Button).toMatch(`import React from 'react';

// eslint-disable-next-line no-unused-vars
function Button(props) {
  return /*#__PURE__*/React.createElement("button", {
    className: "button"
  }, props.children);
}

export default Button;
`);
    expect(styles).toMatch(`.button {
  display: grid;
  transition: all .5s;
  user-select: none;
  background: linear-gradient(to bottom, white, black);
}`);
  });
});
