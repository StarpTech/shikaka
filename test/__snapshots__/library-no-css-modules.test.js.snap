// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library-no-css-modules fixture build with shikaka 1`] = `
"Used script: node cli.js src/index.js --out-dir test/fixtures/library-no-css-modules/dist --root-dir test/fixtures/library-no-css-modules --no-cssmodules

Directory tree:

library-no-css-modules
  dist
    Button.js
    index.js
    styles.css
  package.json
  src
    components
      Button
        index.css
        index.js
    index.js
"
`;

exports[`library-no-css-modules fixture build with shikaka 2`] = `3`;
